src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -162,20 +162,23 @@ public void Dispose()
             {
                 throw new ArgumentNullException("role");
             }
-
             var result = await ValidateRoleInternal(role);
             if (!result.Succeeded)
             {
                 return result;
             }
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(await Store.CreateAsync(role, CancellationToken), role);
+            result = await Store.CreateAsync(role, CancellationToken);
+            using (IdentityLogger.BeginScope(await GetScopeMessageforRoleAsync(role)))
```

Comment:
I had that initially but was reading the BeginScope state from a message since I needed the calling method name. having a method lets the users override the behavior as well